### PR TITLE
Postfix to Prefix conversion in C++

### LINKED ISSUE #5077: Postfix to prefix conversion in C++
## 💥 Proposal

I want to add the code for postfix to prefix conversion in C++. [GSSOC'21]

### Have you read the [Contributing Guidelines on Pull Requests](https://github.com/TesseractCoding/NeoAlgo/blob/master/CONTRIBUTING.md#reporting-new-issues)?

Yes

### PR DESCRIPTION
**Pull request by SAY-droid427**
### Have you read the [Contributing Guidelines on Pull Requests](https://github.com/TesseractCoding/NeoAlgo/blob/master/CONTRIBUTING.md#pull-requests)?

Yes

### Description

I have added the code for postfix to prefix conversion using stacks in C++

### Checklist

- [x] I've read the contribution guidelines.
- [x] I've checked the issue list before deciding what to submit.
- [x] I've edited the `README.md` and link to my code.

## Related Issues or Pull Requests

Fixes https://github.com/TesseractCoding/NeoAlgo/issues/5077

### CODE REVIEW

File: C-Plus-Plus/stack/postfix_to_prefix.cpp [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+* Scan the postfix expression from left to right
+* If the symbol is an operand, push it to the stack
+* If the symbol is an operator, pop two operands from the stack 
+* Create a string by concatenating the two operands and the operator before them. 
+* Push the resultant string back to Stack
+*/
+
+#include<bits/stdc++.h>
+using namespace std;
+void postfixToPrefix(string s)
```
Comment (ankitaggarwal23) [inline C-Plus-Plus/stack/postfix_to_prefix.cpp]: Add a new line between these two

File: C-Plus-Plus/stack/postfix_to_prefix.cpp [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+        cout<<str.top();
+        str.pop();
+    }
+}
+
+//Driver code
+int main()
+{
+    string s;
+    cin>>s;
```
Comment (ankitaggarwal23) [inline C-Plus-Plus/stack/postfix_to_prefix.cpp]: use proper display message

File: C-Plus-Plus/stack/postfix_to_prefix.cpp [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+    postfixToPrefix(s);
+}
+/*
+Time Comlexity:O(n)
+Example:
+Input:
+ABC/-A
+Output: 
+A-A/BC
+*/
```
Comment (ankitaggarwal23) [inline C-Plus-Plus/stack/postfix_to_prefix.cpp]: Add SC as well
Add a space between TC, I/P and O/p
Add a new line at the end

**Review by ankitaggarwal23 — CHANGES_REQUESTED** [review]

File: C-Plus-Plus/stack/postfix_to_prefix.cpp (hunk 1 above)
Comment (SAY-droid427) [reply C-Plus-Plus/stack/postfix_to_prefix.cpp]: Done

File: C-Plus-Plus/stack/postfix_to_prefix.cpp (hunk 2 above)
Comment (SAY-droid427) [reply C-Plus-Plus/stack/postfix_to_prefix.cpp]: Done

File: C-Plus-Plus/stack/postfix_to_prefix.cpp (hunk 3 above)
Comment (SAY-droid427) [reply C-Plus-Plus/stack/postfix_to_prefix.cpp]: Done

**Comment by SAY-droid427** [comment]
@ankitaggarwal23  Please review

File: C-Plus-Plus/stack/postfix_to_prefix.cpp [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+    }
+}
+
+//Driver code
+int main()
+{
+    string s;
+    cout<<"Enter the postfix expression";
+    cin>>s;
+    postfixToPrefix(s);
```
Comment (ankitaggarwal23) [inline C-Plus-Plus/stack/postfix_to_prefix.cpp:48]: Add a display message for output as well. `cout << Prefix Expression:  << endl; `

File: C-Plus-Plus/stack/postfix_to_prefix.cpp [hunk 5]
```diff
@@ -0,0 +1,61 @@
...
+Space Complexity : O(n)
+
+Time Comlexity : O(n)
+
+Example:
+Input:
+ABC/-A
+
+Output: 
+A-A/BC
```
Comment (ankitaggarwal23) [inline C-Plus-Plus/stack/postfix_to_prefix.cpp]: Write in the way your code is producing

**Review by ankitaggarwal23 — CHANGES_REQUESTED** [review]

File: C-Plus-Plus/stack/postfix_to_prefix.cpp (hunk 4 above)
Comment (SAY-droid427) [reply C-Plus-Plus/stack/postfix_to_prefix.cpp:48]: Done

File: C-Plus-Plus/stack/postfix_to_prefix.cpp (hunk 5 above)
Comment (SAY-droid427) [reply C-Plus-Plus/stack/postfix_to_prefix.cpp]: Done

File: C-Plus-Plus/stack/postfix_to_prefix.cpp (hunk 4 above)
Comment (ankitaggarwal23) [reply C-Plus-Plus/stack/postfix_to_prefix.cpp:48]: Where you added that line.

File: C-Plus-Plus/stack/postfix_to_prefix.cpp [hunk 6]
```diff
@@ -0,0 +1,62 @@
...
+        cout<<str.top();
+        str.pop();
+    }
+}
+
+//Driver code
+int main()
+{
+    string s;
+    cout<<"Enter the postfix expression";
```
Comment (ankitaggarwal23) [inline C-Plus-Plus/stack/postfix_to_prefix.cpp]: Use endl or \n as your input is merged with message

**Review by ankitaggarwal23 — CHANGES_REQUESTED** [review]

File: C-Plus-Plus/stack/postfix_to_prefix.cpp (hunk 6 above)
Comment (SAY-droid427) [reply C-Plus-Plus/stack/postfix_to_prefix.cpp]: I had added the line in the function postfixToPrefix(). Now I have moved it to the main().
Added the \n at the end of both the messages.
@ankitaggarwal23

**Review by ankitaggarwal23 — APPROVED** [review]

**Pull request merged**